google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -101,23 +104,35 @@ protected void doStart() {
   @Override
   protected void doStop() {
     messageDispatcher.stop();
-    requestObserver.onError(Status.CANCELLED.asException());
-    notifyStopped();
+
+    lock.lock();
+    try {
+      requestObserver.onError(Status.CANCELLED.asException());
```

Comment:
Unless you think requestObserver could be null, I don't believe it's possible for this to throw an exception.

Reply:
I agree. Though it seems like doing this is a good practice in case we add something else to this in the future? If nothing else, it is recommended by the [docs](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Lock.html)

I guess my view of this is "I know the gun isn't loaded but why point it at my own foot anyway", when the cost of pointing it away is only a few characters.

I don't feel super strongly about this though. Do you?

Reply:
I don't feel strongly, if you do feel free to change it back.